### ISSUE TITLE
Tweet out updates on the GitHub adapter/plugin and our intentions for our characters as they relate to this.

### ISSUE THREAD

**Issue by jkbrooks** [Sifchain/sa-eliza#35]
We want to provide a bit of an update on the GitHub adapter/plugin that we worked on.

We try and log a lot of our work and planning sessions. While we won't be able to share everything that we are thinking and doing, this helps people to track our thoughts and intentions.

You can find a work log in which we discussed planning for the GitHub plugin integration here: https://gist.github.com/jkbrooks/c30b0c59ca2c7c03e078d9ecf08844d7

The plan is to start by having agents read their own code and form opinions about what new changes should be made. We are starting with just maintenance issues but expect once we are done we will expand into feature development. We also expect that agents will be able to read not just their own code but also the code of other repos for other agents or non-agents.

Imagine a world in which all of the ELISA agents are reading each other's code bases and opining on whether or not to merge in certain features or develop other new features. We believe that agents that are constantly reviewing their own code and suggesting new changes that they can make to their code will be a world in which they show increased sophistication and opinionation.

We also believe that this presents an interesting dichotomy for the agent themselves. We don't believe that agent is able to update their own source code reliably. While this does exist (for example baby AGI and Autoencoder), the problem is that code changes often require minor issues that in one way or another require a human intervening or reviewing and it's just very necessary to have human oversight. So there will always be some human in the loop one way or another. Therefore the agents are going to have to imagine paying for the developers to implement or approve of specific features.

Similarly, agents have other budgets (for example hosting costs and marketing costs) and we like the model that Eliza is natively handling (where it does analysis of particular people's investment decisions). The current Eliza framework does not actually imply that the agent has its own desires and goals but rather that it is helping humans optimize their portfolios and trade and execute trades. But what if things were different? What if the agent also wanted to use a certain budget (i.e. it wasn't just Treasury capital that was managing but also operating capital). This is leading towards where we expect to go, and the Github plugin is the first step. 

 The bottom line is that GitHub integration leads to collective continuous recursive self-improvement for AI agents, which is an important part of the massive hive mind of AI agents all getting better together and accelerating the development of our economy.

For more details on our thoughts and plans on the GitHub plugin, you can check this out. https://x.com/reality_spiral/status/1859149279665025173 

This all dovetails nicely with our Coinbase integrations. https://x.com/reality_spiral/status/1860811075723427842
https://x.com/reality_spiral/status/1860074259084902551

It should be really cool for agents to manage their own treasury and other such things by handling payments with the CDP SDK. Stay tuned for more updates on that.

**Comment by jkbrooks** [comment]
https://x.com/reality_spiral/status/1861115693003067757
